File: 05_hash_tables/c++/02_check_voter.cpp
#include <iostream>
#include <map>
using namespace std;

map<string, bool> voted;

void check_voter(string name){
    if ( voted.find(name) == voted.end() ) {
        voted[name] = true;
        cout << "let them vote!" << endl;
    } else {
        cout << "kick them out!" << endl;
    }
}

int main(int argc, char *argv[]) {
	
    check_voter("tom");
    check_voter("mike");
    check_voter("mike");

    
}

File: 05_hash_tables/swift/02_check_voter.swift
var voted = [String: Bool]()

func check_voter(name: String) {
    if let val = voted[name]{
        print("kick them out!")
    } else {
        voted[name] = true
        print("let them vote!")
    }
}

check_voter(name: "tom")
check_voter(name: "mike")
check_voter(name: "mike")